src/kubernetes.cc

```diff
@@ -1042,27 +1042,43 @@ json::value KubernetesReader::FindTopLevelOwner(
 #endif
   const json::Object* metadata = obj->Get<json::Object>("metadata");
 #ifdef VERBOSE
-  LOG(DEBUG) << "FindTopLevelOwner: metadata is " << *metadata;
+  LOG(DEBUG) << "FindTopLevelController: metadata is " << *metadata;
 #endif
   if (!metadata->Has("ownerReferences")) {
 #ifdef VERBOSE
-    LOG(DEBUG) << "FindTopLevelOwner: no owner references in " << *metadata;
+    LOG(DEBUG) << "FindTopLevelController: no owner references in " << *metadata;
 #endif
     return object;
   }
   const json::Array* refs = metadata->Get<json::Array>("ownerReferences");
 #ifdef VERBOSE
-  LOG(DEBUG) << "FindTopLevelOwner: refs is " << *refs;
+  LOG(DEBUG) << "FindTopLevelController: refs is " << *refs;
 #endif
-  if (refs->size() > 1) {
+  std::vector<json::Object> controller_refs;
+  for (const json::value& ref : *refs) {
+    const json::Object* ref_obj = ref->As<json::Object>();
+    if (ref_obj->Has("controller") && ref_obj->Get<json::Boolean>("controller")) {
+      controller_refs.emplace_back(*ref_obj);
+    }
+  }
+#ifdef VERBOSE
+  LOG(DEBUG) << "FindTopLevelController: contoller_refs is " << controller_refs;
+#endif
+  if (controller_refs.size() == 0) {
+#ifdef VERBOSE
+    LOG(DEBUG) << "FindTopLevelController: no controller owner references in " << *metadata;
```

Comment:
Let's fit in 80 characters (line break before the `<<`).

Reply:
done